components/ui-api-layer/docs/resource-quotas-status.md

```diff
@@ -0,0 +1,116 @@
+
+[Here](https://github.com/dtaylor113/origin-web-console/blob/master/app/scripts/services/quota.js#L86) you can find the example implementation of the checking logic in UI.
+
+## The Console calls
+
+The Console calls for the ResourceQuotasStatus automatically. The calls are triggered after: 
+- switching the environment
+- uploading a resource
+- creating a lambda 
+- opening environment's details
+
+## Implementation
+
+This section contains the steps of the ResourceQuotaStatus implementation.
+
+### Calculate the available resources in the Environment
+
+To calculate amount of available resources in the given Environment you must list the ResourceQuotas and loop through them.
+You can get the available amount of resources by calculating the difference from `.spec.hard` and `.status.used` in each ResourceQuota.
+For each resource limit specified in the ResourceQuotas, you must calculate the available amount of resources using the ResourceQuota with the lowest `.spec.hard` value for that resource type.
+
+### Calculate the necessary resources to create the next replica
+
+To calculate how much resources you need to create the next replica, you must sum up the resource usage of all containers in the replica Pod.
+If some resource limit is not specified directly in the replica's template, you must check if any LimitRange is not specifying it.
+In this situation you must add the proper LimitRange limit to the list of necessary resources.
+When the same limit is specified in the LimitRange and in the replica template, the replica's template limit has a priority.
+
+### Check the ReplicaSets
+
+List ReplicaSets in the given Namespace and loop through them. You must check every ReplicaSet which does not reach the number of desired replicas.
+To achieve that, implement the logic which calculates if the ReplicaSet has enough resources to create the next replica.
+When there is not enough resources to create the next replica and the number of desired replicas is not reached, return a ResourceQuotasStatus with a flag set to `true`.
+
+>**NOTE:** The Kubeless functions work with this solution because they use the Deployment and the ReplicaSet underneath.
+
+### Check the StatefulSets
+
+List StatefulSets in the given Namespace and loop through them. You must check every StatefulSet which does not reach the number of desired replicas.
+To achieve that, implement the logic which calculates if the StatefulSet has enough resources to create the next replica.
+If there is not enough resources to create the next replica and the number of replicas is not reached, return a ResourceQuotasStatus with a flag set to `true`.
+
+### End of checking
+
+When both checks passed and the resources usage in the given Environment did not exceed any ResourceQuota limit, return the ResourceQuotasStatus with a flag set to `false`.
```

Comment:
``When both checks have passed and the resources usage in the given Environment did not exceed any ResourceQuota limit, return the ResourceQuotasStatus with a flag set to `false`.``